processor/src/main/java/com/linecorp/decaton/processor/formatter/KeyFormatter.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.processor.formatter;
+
+import java.nio.charset.StandardCharsets;
+
+import com.linecorp.decaton.processor.runtime.internal.TaskKey;
+
+@FunctionalInterface
+public interface KeyFormatter {
+    KeyFormatter DEFAULT = key -> new String(key.array(), StandardCharsets.UTF_8);
+
+    String format(TaskKey key);
```

Comment:
I guess the only place that String-formatted key exposed to users is inside `LoggingContext`.

Then it sounds overkill for me to introduce KeyFormatter just for it.
It makes sense only for users who "consuming non-String topic && using LoggingContext to put Decaton's MDC info", but I'm not sure if it's reasonable to cover such needs.

Also, currently we already have a limitation in key-blocking feature that works properly only for String keys.
Then, introducing key formatter for MDC key formatting (which introduces new interface and new API in processors builder) while Decaton still don't fully support non-String key is bit inconsistent.

The possible option here are:
- A: Simply formatting task's key into MDC key by `new String`
- B: Support arbitrary key-type blocking by using KeyFormatter

I +1 for A for now.
We can consider B again later when we found actual needs.

Reply:
> A: Simply formatting task's key into MDC key by new String

I see. I thought there might be a need that applications consuming a topic with non-String keys print the keys in their own representation for debugging purpose, but it looks that's too much.

Let me just omit `KeyFormatter` for now.